server/Lecture.ts

```diff
@@ -1,31 +1,103 @@
+import Quiz from "./Quiz.ts";
+import Student from "./Student.ts";
 import StudentList from "./StudentList.ts";
-import { WebSocketClient} from "https://deno.land/x/websocket@v0.1.1/mod.ts";
 
 class Lecture {
     tutor: string;
     id: string;
     link: string;
     studentList: StudentList;
     wsc?: WebSocketClient;
+    quizes: Map<String, Quiz>;
 
     constructor(tutor: string) {
         this.tutor = tutor;
         this.id = v4.generate();
         this.link = cryptoRandomString({ length: 7, type: "numeric" });
         this.studentList = new StudentList();
+        this.quizes = new Map;
     }
 
-    setWebSocketClient(wsc: WebSocketClient): void{
+    setWebSocketClient(wsc: WebSocketClient): void {
         this.wsc = wsc;
         this.studentList.on("studentAdded", () => {
-            this.wsc?.send("studentAdded");        
+            this.wsc?.send("studentAdded");
         });
-        this.studentList.on("studentDeleted", () =>{
+        this.studentList.on("studentDeleted", () => {
             this.wsc?.send("studentDeleted");
         });
+        const selectedLecture: Lecture = this;
+        this.wsc.on("message", function (message: string) {
+            const parsed = JSON.parse(message);
+            switch (parsed.event) {
+                case "send_quiz":
+                    selectedLecture.handlerSendQuiz(parsed);
+                    break;
+                default:
+                    console.log("Lecture Websockets: Unexpected type of event")
+
+            }
+        });
+    }
+
+    handlerSendQuiz(parsed: QuizRequestPayload): void {
+        const response: Payload = {
+            event: "quiz_in_progress"
+        }
+        this.wsc?.send(JSON.stringify(response));
+
+        const quiz: Quiz = new Quiz(parsed.data.quiz_id, parsed.data.time_seconds, parsed.data.questions, parsed.data.student_ids);
+        this.quizes.set(quiz.server_id, quiz);
+
+        const selectedStudents: Student[] = this.studentList.asArray().filter((student: Student) => parsed.data.student_ids.includes(student.id));
+
+        const answersAddedHandler = (student: Student, answers: any) => {
+            const serverResponse: ServerQuizResponsePayload = {
+                event: "quiz_answers_added",
+                data: {
+                    quiz_id: quiz.id_from_lecturer,
+                    student_id: student.id,
+                    answers: answers
+                }
+            };
+            this.wsc?.send(JSON.stringify(serverResponse));
+        };
+        quiz.on("answersAdded", answersAddedHandler);
+
+        const quizEndedHandler = (reason: String) => {
+            const serverResponse: QuizEndedPayload = {
+                event: "quiz_ended",
+                data: {
+                    quiz_id: quiz.id_from_lecturer,
+                    reason: reason
+                }
+            };
+            this.wsc?.send(JSON.stringify(serverResponse));
+            if (reason === "quiz_timeout") {
```

Comment:
is this condition about ending quiz after the quiz timeout?

Reply:
yep, it is